src/components/DropdownLogin/DropdownLogin.component.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useReducer } from 'react';
+import { Menu, Dropdown, Avatar, Button, Input } from 'antd';
+import { UserOutlined, LockOutlined } from '@ant-design/icons';
+import useModal from '../../utils/hooks/useModal';
+import ModalComponent from '../Modal/Modal.component';
+import './DropdownLogin.style.css';
+import { useAuth } from '../../providers/Auth';
+import loginApi from '../../utils/services/login';
+import { storage } from '../../utils/storage';
+import { AUTH_IMG_STORAGE_KEY } from '../../utils/constants';
+
+const initialForm = {
+  username: null,
+  password: null,
+};
+
+function formReducer(prevState, { field, value }) {
+  return {
+    ...prevState,
+    [field]: value,
+  };
+}
+
+function DropdownLogin() {
+  const [errorMsg, setErrorMsg] = useState(null);
+  const { isShowing, toggle, setStatusModal } = useModal();
+  const [state, dispatch] = useReducer(formReducer, initialForm);
+  const { login, logout, authenticated } = useAuth();
+
+  const handleSubmit = () => {
+    const { username, password } = state;
+    if (username && password) {
+      loginApi(username, password)
+        .then((result) => {
+          login(result);
+          setStatusModal(false);
+        })
+        .catch(() => {
+          setErrorMsg('Something is wrong in your credentials, Please try again');
+        });
+    } else {
+      setErrorMsg('Please fill the inputs.');
+    }
+  };
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    dispatch({ field: name, value });
+  };
+
+  const triggerLogout = () => {
+    logout();
+  };
+
+  const menu = (
+    <Menu>
+      {!authenticated ? (
+        <Menu.Item key="1" onClick={toggle}>
+          Login
+        </Menu.Item>
+      ) : null}
+      {authenticated ? (
+        <Menu.Item key="1" onClick={triggerLogout}>
```

Comment:
Instead of returning null if you don't wanna render anything you can always use the && operator, like this:
```
{authenticated && (
        <Menu.Item key="1" onClick={triggerLogout}>
          Logout
        </Menu.Item>
      )}
```